[EZ] adjust test to accept training IR input

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132033
* __->__ #131999
* #131995
* #131988

When we do predispatch functional export, sometimes we get harmless additional detach calls. In the new training IR, it actually outputs slightly different (arguable more correct) result.

Differential Revision: [D60348764](https://our.internmc.facebook.com/intern/diff/D60348764/)

tugsbayasgalan committed:
[EZ] adjust test to accept training IR input

When we do predispatch functional export, sometimes we get harmless additional detach calls. In the new training IR, it actually outputs slightly different (arguable more correct) result.

Differential Revision: [D60348764](https://our.internmc.facebook.com/intern/diff/D60348764/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[EZ] adjust test to accept training IR input"


When we do predispatch functional export, sometimes we get harmless additional detach calls. In the new training IR, it actually outputs slightly different (arguable more correct) result.

Differential Revision: [D60348764](https://our.internmc.facebook.com/intern/diff/D60348764/)

[ghstack-poisoned]

bdhirsh commented on test/export/test_export.py:
```diff
@@ -4504,9 +4502,22 @@ def forward(self, x):
+graph():
+    %c_lifted_tensor_0 : [num_users=1] = placeholder[target=c_lifted_tensor_0]
+    %x : [num_users=2] = placeholder[target=x]
+    %lift_fresh_copy : [num_users=1] = call_function[target=torch.ops.aten.lift_fresh_copy.default](args = (%c_lifted_tensor_0,), kwargs = {})
```
side question: the delta in these two graphs (for training vs inference) feels more like an artifact of our tracing infra than something that we tried to do. Should we spend more energy trying to clean up the extra detach() calls that show up in inference IR? (either via trying to avoid them during the tracing process, or in a graph pass)

bdhirsh reviewed (approved):
stop-gap to fix expected failures seems ok

tugsbayasgalan committed:
Update on "[EZ] adjust test to accept training IR input"


When we do predispatch functional export, sometimes we get harmless additional detach calls. In the new training IR, it actually outputs slightly different (arguable more correct) result.

Differential Revision: [D60348764](https://our.internmc.facebook.com/intern/diff/D60348764/)

[ghstack-poisoned]
